File: jvm/java/fizzy/src/main/java/us/yellosoft/fizzy/Fizzy.java
package us.yellosoft.fizzy;

import java.util.stream.IntStream;

/** Example FizzBuzz implementation, parallelized */
public final class Fizzy {
    /** Utility class */
    private Fizzy() {}

    /** Modulo-based FizzBuzz implementation
        @param i Input number
        @return fizzbuzz(i)
    */
    public static String fizzbuzz(final int i) {
        final boolean m = i % 3 == 0;
        final boolean n = i % 5 == 0;

        if (m && n) {
            return "FizzBuzz";
        } else if (m) {
            return "Fizz";
        } else if (n) {
            return "Buzz";
        } else {
            return String.valueOf(i);
        }
    }

    /** CLI entry point
        @param args CLI flags (unused)
    */
    public static void main(final String[] args) {
        final Iterable<String> strings = () -> IntStream
            .range(1, 101)
            .parallel()
            .mapToObj(Fizzy::fizzbuzz)
            .iterator();

        for (final String s: strings) {
            System.out.println(s);
        }
    }
}

File: jvm/kotlin/fizzy/src/main/kotlin/us/yellosoft/fizzy/Fizzy.kt
@file:JvmName("Fizzy")

package us.yellosoft.fizzy

fun fizzbuzz(i: Int): String {
    val m: Boolean = i % 3 == 0
    val n: Boolean = i % 5 == 0

    if (m && n) {
        return "FizzBuzz"
    }

    if (m) {
        return "Fizz"
    }

    if (n) {
        return "Buzz"
    }

    return i.toString()
}

fun main() {
    for (n in 1..100) {
        println(fizzbuzz(n))
    }
}
